### ISSUE TITLE
Add support for compression

### ISSUE THREAD

**Issue by espenhogbakk** [exasol/exasol-driver-ts#30]
There seems to be no support for compressing the data transferred between server and client. I had a look at the internals and i see there are some flags in the `internalConfig` for compression that defaults to `false`. I tried enabling it but seems there is no decompression implemented.

Would be really nice if compression could be added as we see some slow responses related to transfers taking some time.

**Comment by kaklakariada** [comment]
The "compression" flag will compress the messages **after** the login with zlib (see https://github.com/exasol/websocket-api/blob/master/docs/WebsocketAPIV3.md#compression). I assume this would improve performance for large queries or result sets. If this is the case I will add it to the roadmap.

**Comment by espenhogbakk** [comment]
Yes large queries would surely see a benfit of that. And it makes sense for the decompression to be handled by the driver instead of outside i think? And even if I wanted to try to do the decompression outside of the driver there is currently no supported way to configure the driver to use compression since the `compression` is not part of the `Config` interface we can pass into the initializer.

**Comment by pj-spoelders** [comment]
Hi @espenhogbakk 
We've added support for compression (to both the client and pool).
Let us know if you experience any issues. 

Kind regards,
Pieterjan
